[GEOS-10841] Docs for filename and content-disposition vendor options (#6541)
geoserver/geoserver: doc/en/user/source/services/generic_vendor_options.txt

General Vendor Options
----------------------

These vendor options are available for all operations.

content-disposition
^^^^^^^^^^^^^^^^^^^

The ``content-disposition`` parameter directs how a web browser directed to handle returned content. The syntax is::

>    content-dispoition=<dispoition>

Reviewer: ```suggestion
   content-disposition=<disposition>
```
please check spelling; there's several below as well

> Where ``content-dispoition=attachment`` to direct the browser to save the content to disk (even if it is a format such as json or text which it may be wlling to display directly).

Reviewer: ```suggestion
Where ``content-disposition=attachment`` to direct the browser to save the content to disk (even if it is a format such as JSON or plain text which it may be willing to display directly).
```

Author: I will shorten the sentence to just "save to disk".

> Where ``content-dispoition=attachment`` to direct the browser to save the content to disk (even if it is a format such as json or text which it may be wlling to display directly).

Author: ```suggestion
Where ``content-disposition =attachment`` to direct the browser to save the content to disk.
```

> Where ``content-dispoition=inline`` asks the browser to display the content. Note this may present a difficulty when asked to display large content.

Reviewer: ```suggestion
Where ``content-disposition=inline`` asks the browser to display the content. Note this may present a difficulty when asked to display large content.
```

"present a difficulty "  is rather vague... how about 
```suggestion
Where ``content-disposition=inline`` asks the browser to display the content. Note this may present performance issues when asked to display very large content.
```

filename
^^^^^^^^

> The ``filename`` parameter provides a suggested filename when a browser saves a file to Downloads folder. The syntax is::

Reviewer 2: ```suggestion
The ``filename`` parameter provides a suggested filename when a browser saves a file (e.g. to Downloads folder). The syntax is::
```
